pulp_container/app/tasks/sign.py

```diff
@@ -53,19 +53,15 @@ def sign(repository_pk, signing_service_pk, reference, tags_list=None):
     already_signed = []
     for tag in latest_repo_tags:
         tagged_manifest = tag.tagged_manifest
-        if tagged_manifest.media_type in MANIFEST_MEDIA_TYPES.IMAGE:
-            signature_pk = create_signature(
-                tagged_manifest, ":".join((reference, tag.name)), signing_service
-            )
-            added_signatures.append(signature_pk)
-        elif tagged_manifest.media_type in MANIFEST_MEDIA_TYPES.LIST:
-            # parse ML and sign per-arches by digest
+        docker_reference = ":".join((reference, tag.name))
+        signature_pk = create_signature(tagged_manifest, docker_reference, signing_service)
+        added_signatures.append(signature_pk)
+        if tagged_manifest.media_type in MANIFEST_MEDIA_TYPES.LIST:
+            # parse ML and sign per-arches
             for manifest in tagged_manifest.listed_manifests.iterator():
                 # image manifests can be present in multiple ML within the repo
                 if manifest.digest not in already_signed:
-                    signature_pk = create_signature(
-                        manifest, ":".join((reference, manifest.digest)), signing_service
-                    )
+                    signature_pk = create_signature(manifest, docker_reference, signing_service)
```

Comment:
Can we end up in a situation where we will be constantly adding new signatures to the very same manifest listed from different manifest lists? Not sure of any drawback right at the moment. Maybe, it can screw up the number of signatures referenced by manifests?

Reply:
it can be same manifest in other manifest-lists but it will always be referenced by different tag, i.e docker-reference will be different.

Reply:
I was worried about adding new signatures to the same manifest over and over again. But, we will use different tags.

Reply:
if you have manifest A which is used in manifest lists in repos FOO and BAR - that manifest during signing will have different docker-reference: foo:latest ,bar:latest, bar:nightly
But if you trigger sign task again for these repos, yes you will have on top of existing signature created exactly same one, expect for the timestamp. that will be only thing that will be different in the signature payload.

Reply:
since the digest of the signature is calculated based on the payload, this approach brings certain benefits as well as drawbacks, i.e if you in a loop trigger 10 signing tasks on the same repo - it will produce 10 signatures per manifest.